kernel/kernel_drivers.go

```diff
@@ -233,6 +234,13 @@ func EnsureKernelDriversTree(ksnapName string, rev snap.Revision, kernelMount st
 	ksnapDir := ksnapName + "_tmp"
 	if opts.KernelInstall {
 		ksnapDir = ksnapName
+		treeDir := driversTreeDir(ksnapDir, rev)
+		exists, isDir, _ := osutil.DirExists(treeDir)
```

Comment:
could there be a situation where this exists but is empty?

Reply:
It should not be the case, `backend.RemoveKernelSnapSetup` makes sure this is cleaned up on kernel refreshes.

Reply:
but that does RemoveAll that is recursive, so what happens if we reboot in the middle of it? maybe that function to be paranoid should do like some of the backend code in snapstate which I think does rename first+remove?

Reply:
I've deleted my previous comment as I got a bit confused on when this code is reached. In general we will come here on kernel snap refreshes or installation. For the former, `backend.RemoveKernelSnapSetup` should have been called in the undone phase of prepare-kernel-snap if something went wrong in a previous refresh, and should have been retried until it ran fully. For installation, the creation of the folder should have finished successfully when the image is installed, otherwise the image installation would have failed.